core/src/test/java/org/elasticsearch/gateway/MetaDataWriteDataNodesTests.java

```diff
@@ -261,45 +120,51 @@ public void testMetaWrittenWhenIndexIsClosedAndMetaUpdated() throws Exception {
-        assertNotNull(((LinkedHashMap) (getMappingsResponse.getMappings().get("red_index").get("doc").getSourceAsMap().get("properties"))).get("integer_field"));
+        // make sure it was also written on red node although index is closed
+        ImmutableOpenMap<String, IndexMetaData> indicesMetaData = getIndicesMetaDataOnNode(dataNode);
+        assertNotNull(((LinkedHashMap) (indicesMetaData.get(index).getMappings().get("doc").getSourceAsMap().get("properties"))).get("integer_field"));
+        assertThat(indicesMetaData.get(index).state(), equalTo(IndexMetaData.State.CLOSE));
 
-    }
+        // try the same and see if this also works if node was just restarted
+        internalCluster().restartNode(dataNode, new InternalTestCluster.RestartCallback() {
+            public boolean clearData(String nodeName) {
+                return false;
+            }
+        });
+        client().admin().indices().preparePutMapping(index).setType("doc").setSource(jsonBuilder().startObject()
+                .startObject("properties")
+                .startObject("float_field")
+                .field("type", "float")
+                .endObject()
+                .endObject()
+                .endObject()).get();
 
-    private String startDataNode(String color) {
-        return startDataNode(color, createTempDir().toString());
-    }
+        getMappingsResponse = client().admin().indices().prepareGetMappings(index).addTypes("doc").get();
+        assertNotNull(((LinkedHashMap) (getMappingsResponse.getMappings().get(index).get("doc").getSourceAsMap().get("properties"))).get("float_field"));
 
-    private String startDataNode(String color, String newDataPath) {
-        Settings.Builder settingsBuilder = Settings.builder()
-                .put("node.data", true)
-                .put("node.master", false)
-                .put("node.color", color)
-                .put("path.data", newDataPath);
-        return internalCluster().startNode(settingsBuilder.build());
-    }
+        // make sure it was also written on red node although index is closed
+        indicesMetaData = getIndicesMetaDataOnNode(dataNode);
+        assertNotNull(((LinkedHashMap) (indicesMetaData.get(index).getMappings().get("doc").getSourceAsMap().get("properties"))).get("float_field"));
+        assertThat(indicesMetaData.get(index).state(), equalTo(IndexMetaData.State.CLOSE));
 
-    private String startMasterNode() {
-        Settings.Builder settingsBuilder = Settings.builder()
-                .put("node.data", false)
-                .put("node.master", true)
-                .put("path.data", createTempDir().toString());
-        return internalCluster().startNode(settingsBuilder.build());
-    }
+        // finally check that meta data is also written of index opened again
+        client().admin().indices().prepareOpen(index).get();
+        assertTrue(awaitBusy(new Predicate<Object>() {
```

Comment:
can we use assert busy? also, since we have one index, won't it be immediately allocated and the meta data immediately written (before perpareOpen returned). Do we need the busy?